Neos.Flow/Classes/Property/TypeConverter/DenormalizingObjectConverter.php

```diff
@@ -0,0 +1,182 @@
+<?php
+namespace Neos\Flow\Property\TypeConverter;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Property\Exception\TypeConverterException;
+use Neos\Flow\Property\PropertyMappingConfigurationInterface;
+use Neos\Flow\Property\TypeConverterInterface;
+
+final class DenormalizingObjectConverter implements TypeConverterInterface
+{
+    /**
+     * @return array<string>
+     */
+    public function getSupportedSourceTypes()
+    {
+        return ['array', 'string', 'bool', 'int', 'float'];
+    }
+
+    /**
+     * @return string
+     */
+    public function getSupportedTargetType()
+    {
+        return 'object';
+    }
+
+    /**
+     * @param mixed $source the source data
+     * @param string $originalTargetType the type we originally want to convert to
+     * @param PropertyMappingConfigurationInterface $configuration
+     * @return string
+     * @api
+     */
+    public function getTargetTypeForSource($source, $originalTargetType, PropertyMappingConfigurationInterface $configuration = null)
+    {
+        return $originalTargetType;
+    }
+
+    /**
+     * Return the priority of this TypeConverter. TypeConverters with a high priority are chosen before low priority.
+     *
+     * @return integer
+     * @api
+     */
+    public function getPriority()
+    {
+        return 100;
+    }
+
+    /**
+     * @param mixed $source the source data
+     * @param string $targetType the type to convert to.
+     * @return boolean true if this TypeConverter can convert from $source to $targetType, false otherwise.
+     * @api
+     */
+    public function canConvertFrom($source, $targetType)
+    {
+        return self::canConvertFromSourceType(gettype($source), $targetType);
+    }
+
+    /**
+     * @param string $sourceType
+     * @param string $targetType
+     * @return boolean
+     */
+    public static function canConvertFromSourceType(string $sourceType, string $targetType): bool
+    {
+        if (class_exists($targetType)) {
+            switch ($sourceType) {
+                case 'array':
+                    return method_exists($targetType, 'fromArray');
+                case 'string':
+                    return method_exists($targetType, 'fromString');
+                case 'bool':
+                case 'boolean':
+                    return method_exists($targetType, 'fromBoolean');
+                case 'int':
+                case 'integer':
+                    return method_exists($targetType, 'fromInteger');
```

Comment:
Since the types are "officially" `bool` and `int`, why not `fromBool` and `fromInt`?

Less typing, too (think `fn` vs `function`, people seem to like that…)

Reply:
That was already a matter of discussion on the old PR with @bwaidelich mentioning that we do not like abbreviations in method names https://flowframework.readthedocs.io/en/stable/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.html#naming

Reply:
I know… but in this case the potential for confusion can be reduced IMHO, when using the short names.

PHP in itself is inconsistent (_again_), as a type cast can be done using  `(bool)` or `(boolean)` – but a type declaration **must** use `bool`. Same for integers…

This means that the only form that always works is the short form – thus I'd use that for the method names, too. 🤷‍♂️

Reply:
Ok beeing in line with type annotation is a very good argument will adjust the code